plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2456,10 +2501,15 @@ else if (isMetadataColumnId(columnHandle.getId())) {
         }
 
         if (newEnforcedConstraint.equals(table.getEnforcedPredicate())
-                && newUnenforcedConstraint.equals(table.getUnenforcedPredicate())) {
+                && newUnenforcedConstraint.equals(table.getUnenforcedPredicate())
+                && constraint.getPredicateColumns().isEmpty()) {
             return Optional.empty();
         }
-
+        Set<IcebergColumnHandle> icebergColumnHandleSet = constraint.getPredicateColumns()
```

Comment:
I think this should include the columns from enforced and unenforced tuple domains. See the implementation in `DeltaLakeMetadata#applyFilter`

Reply:
On the other hand `constraintColumns` in HiveTableHandle includes only constraint predicate columns. Either way should be okay as long as clearly document in IcebergTableHandle what is included in `constraintColumns`